File: news_server/modules/source/infrastructure/getters/chongbuluo.ts
/**
 * 虫部落新闻源获取器
 * 从 newsnow/server/sources/chongbuluo.ts 迁移
 */
import type { SourceGetter } from '../../domain/getter/getter'
import { NewsItem } from '../../domain/newsitem/entity'
import { defineRSSSource } from './utils'

export const chongbuluoHotGetter: SourceGetter = async () => {
  const baseUrl = 'https://www.chongbuluo.com/'
  const response = await fetch(`${baseUrl}forum.php?mod=guide&view=hot`)
  
  if (!response.ok) {
    throw new Error(`Failed to fetch Chongbuluo: ${response.statusText}`)
  }

  const html = await response.text()
  
  // 使用正则表达式解析 HTML
  const regex = /<tr[^>]*>[\s\S]*?<a[^>]*href="([^"]*)"[^>]*class="[^"]*xst[^"]*"[^>]*>([^<]+)<\/a>/g
  
  const items: Array<{ id: string; title: string; url: string }> = []
  let match

  while ((match = regex.exec(html)) !== null) {
    const url = match[1]
    const title = match[2].trim()
    
    if (url && title) {
      items.push({
        id: baseUrl + url,
        title,
        url: baseUrl + url,
      })
    }
  }

  return items.map((item) =>
    NewsItem.fromEntity({
      id: item.id,
      title: item.title,
      url: item.url,
      extra: {
        hover: item.title,
      },
    })
  )
}

export const chongbuluoLatestGetter = defineRSSSource(
  'https://www.chongbuluo.com/forum.php?mod=rss&view=newthread'
)

File: modules/source/infrastructure/getters/chongbuluo.go
package getters

import (
	"context"
	"regexp"
	"strings"
)

var chongbuluoLinkRe = regexp.MustCompile(`<a[^>]*href="([^"]*)"[^>]*class="[^"]*xst[^"]*"[^>]*>([^<]+)</a>`)

func (r *Registry) chongbuluoHot(ctx context.Context) ([]Item, error) {
	base := "https://www.chongbuluo.com/"
	body, err := r.get(ctx, base+"forum.php?mod=guide&view=hot", nil)
	if err != nil {
		return nil, err
	}
	html := string(body)
	matches := chongbuluoLinkRe.FindAllStringSubmatch(html, -1)
	out := make([]Item, 0, len(matches))
	for _, m := range matches {
		if len(m) < 3 {
			continue
		}
		u := base + m[1]
		title := strings.TrimSpace(m[2])
		out = append(out, Item{ID: u, Title: title, URL: u, Extra: map[string]any{"hover": title}})
	}
	return out, nil
}
